### PR TITLE
Avoid filtering out relationships where entities support filtering

### LINKED ISSUE #26356: Angular: Quick Filter buttons are not generated anymore
##### **Overview of the issue**
Until at least version 8.1.0, buttons for filtering the other side of the entity were generated (initial feature: #18370). This is at least in 8.5.0 not the case anymore, as the corresponding relationships are already filtered out beforehand, so the links are not generated anymore. 

##### **Motivation for or Use Case**

Feature worked before and kind of all of the of the code for that is still in there.

##### **Reproduce the error**

1. Have an application with a ManyToOne / OneToMany /ManyToMany relationship using Angular as frontend.
2. Generate application
3. Expect buttons in the list view pointing from one Entity to the filtered list of the other. (see #18370)
4. Miss out these buttons

##### **Related issues**
Changes in #25140 are causing this, I think. Previously, the list of relationships passed to the client generator were unfiltered apart from the `skipClient` flag.

##### **Suggest a Fix**
get rid of the persistable/eagerload filtering here. Don't know, what effect this has elsewhere, should be tested - or only be changed for the Angular client in doubt:
https://github.com/jhipster/generator-jhipster/blob/eee53c5b07512aaa09d09233476a51da47d9f323/generators/client/support/filter-entities.ts#L25-L29

by the way, the parts in the html of the Angular client are still there though and there you can also see the the contradictionary conditions:

https://github.com/jhipster/generator-jhipster/blob/eee53c5b07512aaa09d09233476a51da47d9f323/generators/angular/templates/src/main/webapp/app/entities/_entityFolder_/list/_entityFile_.component.html.ejs#L178-L191

`&& !persistableRelationship` (Angular code) vs. `persistableRelationship ` (filter-entities.ts)

##### **JHipster Version(s)**
8.5.0

##### **Browsers and Operating System**

- [x] Checking this box is mandatory (this is just to show you read everything)

### PR DESCRIPTION
**Pull request by OmarHawk**
Fix #26356

---

Please make sure the below checklist is followed for Pull Requests.

- [x] [All continuous integration tests](https://github.com/jhipster/generator-jhipster/actions) are green
- [x] Tests are added where necessary
- [x] The JDL part is updated if necessary
- [x] [jhipster-online](https://github.com/jhipster/jhipster-online) is updated if necessary
- [x] Documentation is added/updated where necessary
- [x] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/main/CONTRIBUTING.md) are followed

When you are still working on the PR, consider converting it to Draft (below reviewers) and adding `skip-ci` label, you can still see CI build result at your branch.

### CODE REVIEW

**Comment by OmarHawk** [comment]
This is for discussion whether it makes sense to limit this change to Angular only or to some other condition.

File: generators/client/support/filter-entities.ts [hunk 1]
```diff
@@ -25,7 +25,7 @@ import { type Entity } from '../../base-application/index.js';
 export const filterEntityPropertiesForClient = (entity: Entity): Entity => ({
   ...entity,
   fields: entity.fields.filter(field => !field.skipClient),
-  relationships: entity.relationships.filter(rel => !rel.skipClient && (rel.persistableRelationship || rel.relationshipEagerLoad)),
+  relationships: entity.relationships.filter(rel => !rel.skipClient),
```
Comment (mshima) [inline generators/client/support/filter-entities.ts:28]:
```suggestion
  relationships: entity.relationships.filter(rel => !rel.skipClient && (rel.persistableRelationship || rel.relationshipEagerLoad || rel.otherEntity?.jpaMetamodelFiltering)),
```

I think we should keep the filtering. Backreference is used in very few cases.

Comment (OmarHawk) [reply generators/client/support/filter-entities.ts]: Ok for me. I adjusted it according your suggestion and it works still works :-)

File: generators/client/support/filter-entities.ts [hunk 2]
```diff
@@ -25,7 +25,9 @@ import { type Entity } from '../../base-application/index.js';
 export const filterEntityPropertiesForClient = (entity: Entity): Entity => ({
   ...entity,
   fields: entity.fields.filter(field => !field.skipClient),
-  relationships: entity.relationships.filter(rel => !rel.skipClient && (rel.persistableRelationship || rel.relationshipEagerLoad)),
+  relationships: entity.relationships.filter(
+    rel => !rel.skipClient && (rel.persistableRelationship || rel.relationshipEagerLoad || rel.otherEntity?.jpaMetamodelFiltering),
```
[the remark this answers is not in the document]
Comment (OmarHawk) [reply generators/client/support/filter-entities.ts:29]: wasn't that already possible before, without this change? 

If skipClient = false, then the relationship would have been filtered only if the other conditions produce false (nothing else true).
If skipClient=true, then the relationship would always be filtered out

[the remark this answers is not in the document]
Comment (mshima) [reply generators/client/support/filter-entities.ts:29]: At current state, `@SkipClient(false)` is the same as no annotation.
With the change only undefined skipClient will use default filtering logic.

**Review by mshima — APPROVED** [review]

**Pull request merged**